### ISSUE TITLE
How do we feel about blockinfile?

### ISSUE THREAD

**Issue by danbohea** [hashbangcode/vlad#204]
So I recently gave the [blockinfile](https://github.com/yaegashi/ansible-role-blockinfile) module a whirl on another project and it was a decent enough fit for my needs. Multi-line `lineinfile` tasks are a [PITA](http://stackoverflow.com/questions/24334115/ansible-lineinfile-for-several-lines) and in my opinion that's often a barrier to tasks that should probably be handled by `lineinfile` but often end up being achieved using `template` (or worse) for simplicity.

If Vlad was to adopt it, we'd need to include it at `vlad/library/ansible-role-blockinfile`. Possibly as a git submodule for better flexibility. Thoughts?

**Comment by philipnorton42** [comment]
Agreed, lineinfile is a pain and tends to get a little messy when doing multi line stuff. blockinfile look nice and I don't see any initial problems with adding in custom modules (as long as they work :P).

**Comment by danbohea** [comment]
> as long as they work :P

Absolutely.

OK, I'll look at getting this integrated then and refactor a little code to demonstrate its usage (I'll keep it minor for now in case we decide to back out).

**Comment by philipnorton42** [comment]
Cool. I've slotted it for 1.2 :)

**Comment by danbohea** [comment]
NP. Will hold fire for now.

**Comment by philipnorton42** [comment]
Well, feel free to branch it if you want :)
I'm just keen on getting 1.1 out the door this week :P

**Comment by danbohea** [comment]
> I'm just keen on getting 1.1 out the door this week :P

Yep. That would be cool.
